dask_expr/_collection.py

```diff
@@ -1403,9 +1403,26 @@ def sample(self, n=None, frac=None, replace=False, random_state=None):
             expr.Sample(self, state_data=state_data, frac=frac, replace=replace)
         )
 
-    def rename(self, columns):
+    def rename(self, index=None, columns=None):
+        if index is not None:
+            raise ValueError("Cannot rename index.")
```

Comment:
This is a bit tricky, since when users pass those as args we would then silently use as columns, We could make it keyword only to avoid this maybe?

Merging for now